internal/controllers/machine/machine_controller_phases.go

```diff
@@ -200,6 +193,12 @@ func (r *Reconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.
 	}
 	bootstrapConfig := externalResult.Result
 
+	// If the bootstrap data is populated, set ready and return.
+	if m.Spec.Bootstrap.DataSecretName != nil {
+		m.Status.BootstrapReady = true
```

Comment:
naive question: in a backup and restore scenario, would `externalResult.Paused` on line 198 be true? if that's case could it have side effects of returning early and never setting BootstrapReady to true?

Reply:
I think directly after the actual restore of CRs yes. But the resources have to be unpaused later (otherwise no reconciliation is done at all).

I think it's even more correct that we don't do anything until unpause.

Reply:
I moved pause up a line, so we're processing that before Requeue. I don't think it has any impact in practice, but semantically I think it makes more sense here.